src/kikan/engine.py

```diff
@@ -46,24 +46,27 @@ def _terminate(self):
         Logger.terminate()
 
     def _check_collision(self):
-        for i in range(len(self.game_world.entities)):
-            for j in range(i+1, len(self.game_world.entities)):
-                entity1 = self.game_world.entities[i]
-                entity2 = self.game_world.entities[j]
+        entities = list(self.game_world.entities.values())
+        for i in range(len(entities)):
+            for j in range(i+1, len(entities)):
+                entity1 = entities[i]
+                entity2 = entities[j]
                 if (entity1.position - entity2.position).length() <= 1:
                     self.event_manager.dispatch(
                         entity1, 'collision', [entity2])
                     self.event_manager.dispatch(
                         entity2, 'collision', [entity1])
 
     def _remove_destroyed_entities(self):
-        self.game_world.entities = list(
-            filter(lambda x: not x._is_destroyed, self.game_world.entities))
-        self.game_world.meta_entities = list(
-            filter(lambda x: not x._is_destroyed, self.game_world.meta_entities))
+        for key, value in reversed(self.game_world.entities.items()):
```

Comment:
Why reversed here?

Reply:
Modifying a container while iterating it